include/kernel.h

```diff
@@ -506,9 +506,16 @@ struct _thread_stack_info {
 
 	/* Stack Size - Thread writable stack buffer size. Represents
 	 * the size of the actual area, starting from the start member,
-	 * that should be writable by the thread
+	 * that should be writable by the thread. This includes any TLS storage
+	 * and random offsets applied to the initial stack pointer.
```

Comment:
To clarify, @andrewboie:
You are changing the semantics of 'size', right? This now seems to be 'immutable', while before, the 'size' was adjusted to reflect any reductions due to user-local data storage or round-down adjustments due to randomization, at least for Cortex-M. Or that was generic?

IMO we need to answer the following question:
- Shall the 'stack_info.size' have an 1-1 correspondence to the area that **will eventually** be configured with User RW privileges when the thread enters user mode, or any time the thread switches-in? If so, I wonder if all the 'reduction' due to initial randomization of the stack pointer should be configured as user-accessible, or not... I am OK with both 'yes' or 'no' as long as we specify this.

- Just a note: in Cortex-M we must also round the buffer area down, to make sure it respects the minimum MPU alignment requirements. E.g. for ARMv8-M we have 32-byte alignment requirement. This applies to start and end of the MPU region. We enforce the alignment of the 'start' via K_THREAD_STACK_DEFINE, but the SIZE can by anything, AFAIU. So if the SIZE is not 32-byte aligned, we _anyway_ need to do some round-down of the stack buffer size, and this area cannot be classified as writable by the user thread and shall not be reflected in stack_info.size.

My suggestion to make the specification a bit more clear:

`Thread writable stack buffer size. Represents the size of the actual buffer, starting from the 'start' member, that should be writable by the thread. This comprises of the thread stack area, any area reserved for local thread data storage, as well as any area left-out due to random adjustments applied to the initial thread stack pointer during thread initialization.`

Reply:
My intention is that there is a 1:1 correspondence with the region programmed for User RW access, across all arches. There's no need to make the random region invisible to the user, we can include it and TLS. Your doc suggestion looks great I'll incorporate it.